src/conspiracies/pipeline/config.py

```diff
@@ -16,17 +16,39 @@ class StepConfig(BaseModel):
 
 class PreProcessingConfig(StepConfig):
     input_path: str
+    n_docs: int = -1
     doc_type: str
     metadata_fields: set[str]
     extra: dict = {}
 
 
 class DocProcessingConfig(StepConfig):
+    batch_size: int = 25
+    continue_from_last: bool = True
     triplet_extraction_method: str
 
 
-class CorpusProcessingConfig(StepConfig):
-    pass
+class ClusteringThresholds(BaseModel):
+    n_neighbors: int = 15
+    min_cluster_size: int = 3
+    min_samples: int = 3
+    min_topic_size: int = 5
+
+    @classmethod
+    def auto(cls, n_triplets: int):
+        factor = n_triplets / 1000
+        thresholds = cls()
+        thresholds.n_neighbors = int(factor * 5)
+        thresholds.min_cluster_size = int(factor) + 1
+        thresholds.min_samples = int(factor) + 1
+        thresholds.min_topic_size = int(factor * 2) + 1
+        return thresholds
+
+
+class CorpusProcessingConfig(StepConfig, ClusteringThresholds):
+    dimensions: int = -1
+    embedding_model: str = None
+    auto_thresholds: bool = True
```

Comment:
Oh so it contains the other to and is then flat?
I would probably just make it nested 

```suggestion
class CorpusProcessingConfig():
    dimensions: int = -1
    embedding_model: str = None
    auto_thresholds: bool = True
    step_config: StepConfig
    ...
```

Reply:
Probably also a matter of preference. While the nested structure has its merits, it makes the configuration file layout more complex. Then you would have something like:
```toml
[corpusprocessing]
dimensions = -1
embedding_model = "PATH_OR_NAME"
[corpusprocessing.step_config]
enabled = false
[corpusprocessing.clustering_thresholds]
n_neighbors = 15
min_cluster_size = 3
min_samples = 3
min_topic_size = 5
```

I actually moved back and forth between strategies with the thresholds before committing. It's about striking the right balance.

Reply:
Yep can imagine. Whatever you prefer